salt/utils/versions.py

```diff
@@ -140,26 +164,92 @@ def warn_until(version,
         )
 
     if _dont_call_warnings is False:
-        def _formatwarning(message,
-                           category,
-                           filename,
-                           lineno,
-                           line=None):  # pylint: disable=W0613
-            '''
-            Replacement for warnings.formatwarning that disables the echoing of
-            the 'line' parameter.
-            '''
-            return '{0}:{1}: {2}: {3}\n'.format(
-                filename, lineno, category.__name__, message
+        with _patched_format_warning():
+            warnings.warn(
+                message.format(version=version.formatted_version),
+                category,
+                stacklevel=stacklevel
+            )
+
+
+def _get_utcnow_date():
```

Comment:
This can also be solved by making this and other time methods injectable dependencies to `warn_until_date`. This way tests can use a different version of the method without needing a separate mockable method in this module.

Reply:
Let's address that in a future PR.